### PR TITLE
feat(fines): fines now sends out notifs and fine admins now gets an email

### LINKED ISSUE #364: Varsler for botsystemet
Som ny Botsjef i en gruppe ønsker jeg å bli varslet om min nye stilling. Jeg ønsker også å motta info om hva jeg nå har mulighet til og hvordan. Send både varsel på nettsiden og epost.

Som medlem i en gruppe som mottar en bot ønsker jeg å bli varslet om dette. Ettersom jeg får mange bøter ønsker jeg ikke varsel på epost, kun på nettsiden.

### PR DESCRIPTION
**Pull request by Her0elt**
## Proposed changes
Added notifications when creating a fine, and if a user is promoted to fines admin they are sendt an email with information about their new role

Issue number: closes #364 

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] The fixtures have been updated if needed (for migrations)
- [x] Pull request title follows [conventional commits](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) (`type(scope): description`)
- [x] Format and lint (`make format` and `make flake8`) was run locally without failures

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,8 @@
 ---
 
 ## Neste versjon
+- ✨ **Varsler Botsystem**. Medlemmer får nå varsler hvis de får en bot
+- ✨ **Botsjef **. Hvis man får lagt til som botsjef så får man nå en mail om sin nye rolle
```
Comment (olros) [inline CHANGELOG.md:17]:
```suggestion
- ✨ **Botsystem**. Medlemmer får nå et varsl når de får en ny bot
- ✨ **Botsjef**. Nye botsjefer får nå informasjon om sin nye roller på varsel og epost.
```

File: app/group/models/fine.py [hunk 2]
```diff
@@ -40,9 +40,22 @@ def clean(self):
...
+        if not self.id:
+            self.notify_user()
+
+    def notify_user(self):
+        from app.util.notifier import Notify
+
+        Notify(
+            [self.user], f"Du har fått en bot i gruppen {self.group.name}"
+        ).send_notification(
+            description=f"{self.created_by.first_name} {self.created_by.last_name} har gitt deg {self.amount} bøter for å ha brutt {self.description} i gruppen {self.group.name}",
```
Comment (olros) [inline app/group/models/fine.py:52]:
```suggestion
            description=f"{self.created_by.first_name} {self.created_by.last_name} har gitt deg {self.amount} bøter for å ha brutt paragraf \"{self.description}\" i gruppen {self.group.name}",
```

File: app/group/models/group.py [hunk 3]
```diff
@@ -34,17 +35,54 @@ class Meta:
...
+            )
+            .add_paragraph(
+                f"Du har blitt satt som ansvarlig for bøter i gruppen {self.name}, du kan nå administrere bøtene for denne gruppen."
+            )
+            .add_button("Gå til gruppe siden", f"{self.website_url}",)
+            .generate_string()
+        ).send_notification(
+            description=f"Du har blitt satt som ansvarlig for bøter i gruppen {self.name}, du kan nå administrere bøtene for denne gruppen.",
+            link=f"/grupper/{self.slug}/boter/",
+        )
```
Comment (olros) [inline app/group/models/group.py:57]: Her er en litt mer utdypende tekst:

```suggestion
        
        description = [
            f"Hei! Lederen av {self.name} har gjort deg til botsjef i gruppen. I botsystemet i gruppen kan alle medlemmer melde inn bøter på en eller flere andre medlemmer. Gruppen kan selv velge hvor mye en bot er verdt og hvordan bøter skal godkjennes og betales.",
            "Som botsjef kan du, sammen med leder av gruppen, redigere lovverket og bøter (godkjenne, markere som betalt, endre antall og slette). Medlemmene kan ikke opprette bøter før det er minst én lov i lovverket. Du og alle medlemmene kan se en oversikt over alle bøter, samt filtrere på om boten er godkjent, betalt og per medlem.",
            "Lykke til!"
        ]

        Notify(
            [self.fines_admin], f"Du har blitt botsjef for gruppen {self.name}"
        ).send_email(
            MailCreator(f"Du har blitt botsjef for gruppen {self.name}")
            .add_paragraph(description[0})
            .add_paragraph(description[1])
            .add_paragraph(description[2])
            .add_button("Gå til gruppesiden", self.website_url)
            .generate_string()
        ).send_notification(
            description=" \n".join(description), link=f"/grupper/{self.slug}/boter/",
        )
```

File: app/group/models/group.py [hunk 4]
```diff
@@ -34,17 +35,54 @@ class Meta:
     def __str__(self):
         return f"{self.name}"
 
+    def noftify_fines_admin(self):
```
Comment (olros) [inline app/group/models/group.py:38]: Skrivefeil ;)
```suggestion
    def notify_fines_admin(self):
```

File: app/group/models/group.py [hunk 5]
```diff
@@ -34,17 +35,54 @@ class Meta:
...
+        )
+
+    def check_fine_admin(self):
+        return (
+            self.fines_admin is not None
+            and not len(
+                Group.objects.filter(slug=self.slug, fines_admin=self.fines_admin)
+            )
+            > 0
+        )
```
Comment (olros) [inline app/group/models/group.py:66]: Denne sjekker vel kun om man går fra å ikke ha en botsjef til å ha botsjef? Må varsle botsjef om man bytter fra en botsjef til en annen også.

(Og bør vel heller bruke `.exists` istedenfor `len() > 0`)

File: app/group/models/group.py [hunk 6]
```diff
@@ -34,17 +35,54 @@ class Meta:
...
+
         super().save(*args, **kwargs)
 
     @classmethod
     def check_context(cls, request):
         return request.parser_context.get("kwargs", {}).get("slug", None) is not None
 
+    @property
+    def website_url(self):
+        return (f"{settings.WEBSITE_URL}/grupper/{self.slug}/",)
```
Comment (olros) [inline app/group/models/group.py:84]: Hvorfor et tuple og ikke bare en string her?

**Review by olros — CHANGES_REQUESTED** [review]

File: app/group/models/group.py (hunk 6 above)
Comment (Her0elt) [reply app/group/models/group.py:84]: Idk formating la det må må nok ha glømt å ta bort en komma

**Review by olros — APPROVED** [review]
Meget flott! Fin løsning på sjekk av om ny botsjef 🚀

**Pull request merged**